hack/local-setup.sh

```diff
@@ -147,16 +147,23 @@ installGatewayAPI() {
 
 deployKuadrant(){
   clusterName=${1}
-  kubectl config use-context kind-${clusterName}
 
-  echo "Installing Kuadrant in ${clusterName}"
-  ${KUSTOMIZE_BIN} build config/kuadrant | kubectl apply -f -
   kubectl config use-context kind-mctc-control-plane
   echo "Installing Redis in kind-mctc-control-plane"
-  ${KUSTOMIZE_BIN} build ${REDIS_KUSTOMIZATION_DIR} | kubectl apply -f - 
+  ${KUSTOMIZE_BIN} build ${REDIS_KUSTOMIZATION_DIR} | kubectl apply -f -
+
   kubectl config use-context kind-${clusterName}
+  echo "Installing Kuadrant in ${clusterName}"
+  ${KUSTOMIZE_BIN} build config/kuadrant | kubectl apply -f -
   echo "Configuring Limitador in ${clusterName}"
   ${KUSTOMIZE_BIN} build ${LIMITADOR_KUSTOMIZATION_DIR} | kubectl apply -f -
+  echo "Waiting for Kuadrant deployments to be ready..."
+  kubectl -n kuadrant-system wait --timeout=300s --for=condition=Available deployments --all
+  echo "Adding Kuadrant CR in ${clusterName}"
+  ${KUSTOMIZE_BIN} build config/kuadrant/cr | kubectl apply -f -
+  # This is adding the kuadrant CR before any gateways are created which is currently a known issue with kuadrant.
+  # After adding a gateway is will necessary to force a reconcile of the kuadrant CR, this can be done by running:
+  # kubectl annotate kuadrant/mctc -n kuadrant-system updateme=`date +%s` --overwrite
 }
```

Comment:
@R-Lawton I added the creation of the kuadrant CR here. There is an issue with kuadrant in that it only reconciles the gateways when you create or update the kuadrant CR, which is the reason it works in the M2 walkthough as we do it in that order, but it shouldn't have to be done that way. Until there is a fix, we will just have to remember that when we create a gateway that we force a reconcile of the kuadrant cr on the workload clusters. @david-martin fyi

@maleck13 This got me thinking, should we be creating the kuadrant CR as part of HCG and syncing these out to the clusters? re. https://github.com/Kuadrant/architecture/pull/5 seems we could eventually have one per tenant, and would allow us in the short term to force an update of the kuadrant CR (update an annotation) every time a gateway is created in HCG.

Reply:
> Until there is a fix
Is there an upstream issue to track in the kuadrant operator?